code/__HELPERS/colors.dm

```diff
@@ -64,3 +64,11 @@
 
 #define RANDOM_COLOUR (rgb(rand(0,255),rand(0,255),rand(0,255)))
 
+
+///Given a hexcolor (#FFFFFF), returns the inverted hexcolor.
+/proc/invert_hex(hexcolor)
+	var/list/old_rgb = hex2rgb(hexcolor)
```

Comment:
you should use rgb2num here
why do we have hex2rgb you may ask? mothblocks couldn't find rgb2num
https://secure.byond.com/docs/ref/#/proc/rgb2num

Reply:
Also is rgb space the best thing to work in here? what's the goal of this inversion

Reply:
to get the color on the other end of the color wheel. Also, if rgb2num and hex2rgb are the same functionaly, what's the problem? hex2rgb sounds more intuitive.

Reply:
> Also is rgb space the best thing to work in here? what's the goal of this inversion

IIRC, it was to be able to properly invert the value of a hex. The reason was that there's a pAI that has an animation where the color of the light goes from the normal to the total opposite of it.

Reply:
ok this'll work then. Typically operating in rgb space causes headaches, but this is fine.




> 
> 
> to get the color on the other end of the color wheel. Also, if rgb2num and hex2rgb are the same functionaly, what's the problem? hex2rgb sounds more intuitive.

rgb2num is an internal proc, but hex2rgb is something we've had to roll for ourselves. There's the same thing functionally, and I'd prefer not to have 2 things to do one thing. I've been meaning to get rid of hex2rgb because of this, just haven't got around to it yet